src/pages/index.js

```diff
@@ -0,0 +1,236 @@
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  console.log(card);
+  return card.getView();
+}
+
+function handleProfileEditSubmit({ title, description }) {
+  editModal.setLoading(true);
+  api
+    .setUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: title, about: description });
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.setLoading(false);
+    });
+}
+
+function handleAddCardSubmit(name, url) {
+  addModal.setLoading(true);
+  api
+    .addCard(name, url)
+    .then((data) => {
+      const card = renderCard(data);
+      section.addItem(card);
+      addModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
+}
+
+function handleAvatarSubmit(url) {
+  profileAvatarPopUp.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.getAvatar(data);
+      profileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopUp.setLoading(false);
+    });
+}
+
+function handleDeleteCard(card) {
+  cardDeletePopUp.open();
+  cardDeletePopUp.setSubmitAction(() => {
+    cardDeletePopUp.setLoading(true, "Deleting...");
+    api
+      .deleteCard(card.id)
+      .then(() => {
+        card.handleDeleteCard();
+        cardDeletePopUp.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopUp.setLoading(false, "Yes");
+      });
+  });
+} //fixed the delete!!!
+
+function handleLike(cardInstance) {
+  if (cardInstance.isLiked) {
+    api
+      .dislikeCard(cardInstance.id)
+      .then(() => {
+        cardInstance.handleLikeIcon();
+        cardInstance.isLiked = false;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  if (!cardInstance.isLiked) {
+    api
+      .likeCard(cardInstance.id)
+      .then(() => {
+        cardInstance.handleLikeIcon();
+        cardInstance.isLiked = true;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+const profileFormValidator = new FormValidator(config, profileEditForm);
+profileFormValidator.enableValidation();
+
+const addFormValidator = new FormValidator(config, addCardForm);
+addFormValidator.enableValidation();
+
+const avatarFormValidator = new FormValidator(config, profileAvatarForm);
+avatarFormValidator.enableValidation();
+
+const profileAvatarPopUp = new PopUpWithForm(
+  "#avatar-modal",
+  handleAvatarSubmit,
+  profileAvatarButton
+);
+
+profileAvatarButton.addEventListener("click", () => {
+  profileAvatarPopUp.open();
+});
+profileAvatarPopUp.setEventListeners();
+
+const editModal = new PopUpWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+
+profileEditButton.addEventListener("click", () => {
```

Comment:
Needs correction: you should fill the profile form before opening it

Reply:
To clarify, when the edit form opens, it shows name and description, are you meaning it should show the current info?

Reply:
The form war empty, but it should be filled with the current info

Reply:
Sounds good, thank you!